### PR TITLE
User Management: Tracking events

### LINKED ISSUE #87608: User management: Improve tracking
sso.php file makes good use of tracking ( `$tracking->record_user_event` ).

We should expand our tracking of the user management changes we introduced.

### PR DESCRIPTION
**Pull request by escapemanuele**
Fixes https://github.com/Automattic/wp-calypso/issues/87608

## Proposed changes:

Add tracking events for User management, `/wp-admin/users.php`, SSO changes.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

* Sync to Atomic site
* Navigate to `/wp-admins/users.php`.
* Trigger the SSO user actions (Revoke Invite, Send Invite) and add a new user.
* Navigate to the Tracks Live View (MC: `/tracks/live/`) , filter by your username, and verify the events are being tracked correctly.

### CODE REVIEW

**Comment by agrullon95** [comment]
I updated the PR to only rely on a single instance of the tracking object and added tracking to other areas where we have relevant user actions.

The events are not properly tracked on my WoA as of now (more context: p1708469267317199/1708434623.948299-slack-CDLH4C1UZ), but I'm still looking into a solution.

File: projects/plugins/jetpack/modules/sso/class.jetpack-sso-user-admin.php [hunk 1]
```diff
@@ -624,9 +769,23 @@ public function send_wpcom_mail_user_invite( $errors, $update, $user ) {
...
+					self::$tracking->record_user_event(
+						'sso_user_new_user_invite_send',
+						array(
+							'success' => 'false',
+						)
+					);
+				} else {
+					$custom_message_sent = isset( $new_user_request['message'] ) ? 'true' : 'false';
+					self::$tracking->record_user_event(
+						'sso_user_new_user_invite_send',
```
Comment (heavyweight) [inline projects/plugins/jetpack/modules/sso/class.jetpack-sso-user-admin.php]: Could we use a different name for this event or this one is ok? 
Its a bit confusing when you read it

File: projects/plugins/jetpack/modules/sso/class.jetpack-sso-user-admin.php [hunk 2]
```diff
@@ -247,70 +318,117 @@ public function revoke_wpcom_invite( $invite_id ) {
...
 			if ( ! current_user_can( 'promote_users' ) ) {
+				$error        = 'invalid-revoke-permissions';
 				$query_params = array(
 					'jetpack-sso-invite-user'  => 'failed',
-					'jetpack-sso-invite-error' => 'invalid-revoke-permissions',
+					'jetpack-sso-invite-error' => $error,
 					'_wpnonce'                 => $nonce,
 				);
 
+				self::$tracking->record_user_event(
```
Comment (heavyweight) [inline projects/plugins/jetpack/modules/sso/class.jetpack-sso-user-admin.php]: I think we can remove this

**Review by heavyweight — APPROVED** [review]
I can verify that the events are sent out

**Pull request merged**